tasks/src/tasks/select.rs

```diff
@@ -0,0 +1,194 @@
+use std::num::NonZeroU32;
+
+use imap_types::{
+    command::CommandBody,
+    flag::{Flag, FlagPerm},
+    mailbox::Mailbox,
+    response::{Code, Data, StatusBody, StatusKind},
+};
+use tracing::{error, warn};
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug, Default)]
+pub struct SelectData {
```

Comment:
Should this struct move to `imap-types`?

Reply:
Not sure... maybe. Hard to tell where the boundaries are.

Reply:
Thoughts: Ideally, `imap-types` should have only types that are definitely valid by spec. When we add `SelectData`, we would need to do it using `Option` for all fields, which, strictly speaking, is not valid. So, one option would be to call it `SelectData{Unvalidated,Raw}`, etc. But, I think the answer could also be: Keep it in an upper layer.

Reply:
> I think the answer could also be: Keep it in an upper layer

But then what would be the return type of this task? And where would be its definition? 

> So, one option would be to call it `SelectData{Unvalidated,Raw}`

It could also be a giant tuple, but I find it uggly. Let's go for `SelectDataUnvalidated`?